code/obj/machinery/phone/phone.dm

```diff
@@ -119,13 +121,17 @@ TYPEINFO(/obj/machinery/phone)
 			if(src.labelling)
 				return
 			src.labelling = TRUE
-			var/t = input(user, "What do you want to name this phone?", null, null) as null|text
+			var/t = tgui_input_text(user, "What do you want to name this phone?", null, null, max_length = 50)
+			src.labelling = FALSE
 			t = sanitize(html_encode(t))
-			if(t && length(t) > 50)
+			if(!t)
 				return
-			if(t)
-				src.phone_id = t
-			src.labelling = FALSE
+			if(!in_interact_range(src, user))
+				return
+			if(length(t) > 50)
+				return
```

Comment:
```suggestion
```
Fun fact actually, you don't need this since `tgui_input_text` validates the length server side already (even though I'm pretty sure most of our code still checks again), and putting the check here might break with html encoding modifying the length.

Reply:
oh cool, I didn't realize that!